src/utils.ts

```diff
@@ -1299,3 +1299,8 @@ export const COSMOS_DB_MSG =
 export function isHostMatch(match: RegExp, host?: string): boolean {
   return host && match.test(host.toLowerCase()) ? true : false;
 }
+
+export async function mayAbort(signal?: AbortSignal) {
+  if (signal == null) return new Promise(() => null); // never ending story
+  return new Promise((_, reject) => signal.addEventListener('abort', reject));
+}
```

Comment:
can this just be consolidated?
```suggestion
export async function mayAbort(signal?: AbortSignal) {
  return new Promise((_, reject) => signal?.addEventListener('abort', reject));
}
```

Reply:
(also) - would you mind adding a nice doc comment?  I think good doc comments would on our utilities would be really nice for future devs

Reply:
((also)) should we test this utility?

Reply:
Added tests and improved the function, we also needed a throw if the signal already aborted condition.